azurelinuxagent/ga/update.py

```diff
@@ -982,6 +986,34 @@ def _ensure_firewall_rules_persisted(dst_ip):
             message=msg,
             log_event=False)
 
+    @staticmethod
+    def _add_dns_tcp_firewall_rule_if_not_enabled(dst_ip):
+
+        # Helper to execute a run command, returns True if no exception else returns False
+        def _execute_run_command(command):
+            try:
+                shellutil.run_command(command)
+                return True
+            except Exception:
+                return False
+
+        # "-C" checks if the iptable rule is available in the chain. It throws an exception if the ip table rule doesnt exist
+        drop_rule = _get_firewall_drop_command("-w", "-C", dst_ip)
+        if not _execute_run_command(drop_rule):
+            return
+        else:
+            accept_non_root = _get_firewall_accept_dns_tcp_request_command("-w", "-C", dst_ip)
+            if not _execute_run_command(accept_non_root):
```

Comment:
Can you add a log statement stating `Not adding DNS firewall rule since it already exists` or something. Thanks!